apollographql/apollo
# Best practices guides cleanup

This PR reflects all the cleanup resulting from the platform docs launch.
- Authentication guide moves to Apollo Server
- Schema design guide moves to Apollo Server, versioning content moves to `Validating schema changes` platform article
- Testing guide moves to Apollo Client
- File uploads is redirected to Prosper's blog post
- State mgmt redirects to Apollo Client local state article

I also moved the Apollo Config article to the platform section.

Related PRs that should be merged before this one:
https://github.com/apollographql/apollo-client/pull/4143
https://github.com/apollographql/apollo-server/pull/1967

cc @abernix
